### ISSUE TITLE
Multi-Step building for docker (front-end)

### ISSUE THREAD

**Issue by supaidaman** [urban-toolkit/utk#70]
Currently, the dockerfile for the react frontend is running the appplication in de mode(start:web). That usally brings a bunch of dependencies that are not needed to the application itself. We can make that docker image smaller by using the following steps: 

- Using a multi-stage build, so that only what's relevant to the project itself is bundled
- Trying to use a tool like webpack or esbuild to output the file as a minified file.

I have a sample dockerfile for the type of application I usually have here (I haven't tried changing at the code as of now to see how it will behave):

```
FROM node18-alpine3:latest AS base

WORKDIR /home/appuser/

FROM base as development

COPY   package.json ./
COPY   package-lock.json ./
COPY   .npmrc ./

RUN    npm  install --omit=dev
RUN    cp -R node_modules /tmp/node_modules

RUN npm install
COPY . .
RUN pwd
RUN ls -la


FROM development as builder

RUN npm run build
RUN pwd
RUN ls -la

FROM base as release

COPY  --from=builder /tmp/node_modules ./node_modules
COPY  --from=builder /home/appuser/dist ./dist


CMD [ "node", "./dist/app.js" ]
USER appuser

```
Basically, this makes the final app to have only the relevant production node_modules files and the outputed dist files.
